csharp/ql/lib/semmle/code/csharp/Type.qll

```diff
@@ -1126,7 +1124,10 @@ class ArglistType extends Type, @arglist_type {
  * A type that could not be resolved. This could happen if an indirect reference
  * is not available at compilation time.
  */
-class UnknownType extends Type, @unknown_type { }
+class UnknownType extends Type, @unknown_type {
+  /** Holds if this is the canonical uknown type, and not a type that failed to extract properly. */
+  predicate isCanonical() { types(this, _, "<unknown type>") }
```

Comment:
I don't think it changes anything, but we could be more explicit here by specifying `31` instead of `_`.

Reply:
I would rather not, as it is already implied from the `extends @unknown_type`.